### ISSUE TITLE
使用推荐的数据进行本文检测一直卡主不动

### ISSUE THREAD

**Issue by admin0528** [PaddlePaddle/PaddleOCR#1225]
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/root/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ocr/Desktop/OCR/projects/PaddleOCR/ppocr/modeling/losses/det_basic_loss.py:100
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/root/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ocr/Desktop/OCR/projects/PaddleOCR/ppocr/modeling/losses/det_basic_loss.py:103
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/root/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ocr/Desktop/OCR/projects/PaddleOCR/ppocr/modeling/losses/det_basic_loss.py:113
The behavior of expression A - B has been unified with elementwise_sub(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_sub(X, Y, axis=0) instead of A - B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/root/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ocr/Desktop/OCR/projects/PaddleOCR/ppocr/modeling/losses/det_basic_loss.py:113
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
2020-11-23 10:21:24,677-INFO: If regularizer of a Parameter has been set by 'fluid.ParamAttr' or 'fluid.WeightNormParamAttr' already. The Regularization[L2Decay, regularization_coeff=0.000000] in Optimizer will not take effect, and it will only be applied to other Parameters!
2020-11-23 10:21:25,741-INFO: places would be ommited when DataLoader is not iterable
W1123 10:21:25.774652  7668 device_context.cc:320] Please NOTE: device: 0, CUDA Capability: 75, Driver API Version: 11.1, Runtime API Version: 10.0
W1123 10:21:25.776715  7668 device_context.cc:328] device: 0, cuDNN Version: 7.6.
2020-11-23 10:21:28,152-INFO: Loading parameters from ./pretrain_models/MobileNetV3_large_x0_5_pretrained/...
2020-11-23 10:21:28,229-INFO: Finish initing model from ./pretrain_models/MobileNetV3_large_x0_5_pretrained/
2020-11-23 10:21:28,229-INFO: During the training process, after the 800th iteration, an evaluation is run every 200 iterations
3 640 640
3 640 640
./train_data/icdar2015/text_localization/train_icdar2015_label.txt

程序卡在这个位置一直不动，请问是什么问题，训练的步骤都是照着文档来的，det_mv3_db_v1.1.yml和det_db_icdar15_reader.yml没有做修改 。

**Comment by Haowei-Huang** [comment]
我也是卡主，请问楼主怎么解决的

**Comment by admin0528** [comment]
> 我也是卡主，请问楼主怎么解决的

修改PaddleOCR/ppocr/data/reader_main.py配置文件中第75行
return paddle.reader.multiprocess_reader(readers, False)
默认读取图片大小为320张，我是6G显存不够导致显存占满程序崩溃，添加queue_size=160，或者依次减少
return paddle.reader.multiprocess_reader(readers, False,queue_size=160)

**Comment by Haowei-Huang** [comment]
> > 我也是卡主，请问楼主怎么解决的
> 
> 修改PaddleOCR/ppocr/data/reader_main.py配置文件中第75行
> return paddle.reader.multiprocess_reader(readers, False)
> 默认读取图片大小为320张，我是6G显存不够导致显存占满程序崩溃，添加queue_size=160，或者依次减少
> return paddle.reader.multiprocess_reader(readers, False,queue_size=160)

好的，谢谢
